### PR TITLE
feat: setup improve for testing

### PR DESCRIPTION
**Pull request by Lanttcat**
- feat: split wallet app entries
- chore: type
- chore: type
- feat: user guide
- chore: solve conflict
- Revert "chore: solve conflict"
- chore: solve conflict
- feat: should open twitter when user click agree button on welcome page
- feat: be opened social network should determined by the caller
- feat: merge develop
- feat: recommend feature app
- chore: update event
- chore: persona connect
- fix: little fix
- fix: dialog title text center
- fix: style
- chore: add comment
- fix: ui style
- chore: ui style

## Description

Closes #MF-415 #MF-416

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/shared/src/contexts/components/InjectedDialog.tsx [hunk 1]
```diff
@@ -139,7 +149,7 @@ export function InjectedDialog(props: InjectedDialogProps) {
                             <Typography className={dialogTitleTypography} display="inline" variant="inherit">
                                 {title}
                             </Typography>
-                            {titleTail}
+                            <div className={dialogTitleTail}>{titleTail}</div>
```
Comment (guanbinrui) [inline packages/shared/src/contexts/components/InjectedDialog.tsx]:
```suggestion
                            <Typography className={dialogTitleTail}>{titleTail}</Typography>
```

File: packages/mask/src/components/GuideStep/index.tsx [hunk 2]
```diff
@@ -119,6 +119,12 @@ export default function GuideStep({
         if (disabled) return
         const open = +lastStep === step
         setOpen(open)
+
+        if (!open) return
+        const url = new URL(location.href)
+        if (url.pathname === '/home') return
```
Comment (guanbinrui) [inline packages/mask/src/components/GuideStep/index.tsx]: It should not suppose the users are on Twitter.

Comment (yanzhihong23) [reply packages/mask/src/components/GuideStep/index.tsx]: Only Twitter has user guide currently.

Comment (septs) [reply packages/mask/src/components/GuideStep/index.tsx]:
```suggestion
        if (location.pathname === '/home') return
```

File: packages/mask/src/social-network/ui.ts [hunk 3]
```diff
@@ -226,6 +228,19 @@ export async function activateSocialNetworkUIInner(ui_deferred: SocialNetworkUI.
     }
 }
 
+export async function loadSocialNetworkUIs(): Promise<SocialNetworkUI.Definition[]> {
```
Comment (Jack-Works) [inline packages/mask/src/social-network/ui.ts:231]: Why you're going to load every SNS Adaptors?

File: packages/mask/src/components/InjectedComponents/ToolboxUnstyled.tsx [hunk 4]
```diff
@@ -82,8 +82,56 @@ export interface ToolboxHintProps {
...
+    } = props
+    const { classes } = useStyles()
+    const { t } = useI18N()
+    const { openDialog } = useRemoteControlledDialog(WalletMessages.events.ApplicationDialogUpdated)
+    return (
+        <GuideStep step={1} total={4} tip={t('user_guide_tip_1')}>
+            <Container>
+                <ListItemButton onClick={openDialog}>
+                    <img
+                        src={new URL('../../plugins/EVM/assets/maskwallet.png', import.meta.url).toString()}
```
Comment (Jack-Works) [inline packages/mask/src/components/InjectedComponents/ToolboxUnstyled.tsx]: please move this image out of the plugin

File: packages/dashboard/src/pages/Welcome/hooks/useSetupSocialNetwork.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import { useAsyncFn } from 'react-use'
+import { Services } from '../../../API'
+
+export function useSetupSocialNetwork() {
+    return useAsyncFn(Services.SocialNetwork.setupSocialNetwork)
```
Comment (Jack-Works) [inline packages/dashboard/src/pages/Welcome/hooks/useSetupSocialNetwork.ts]: Why? Can't you just call it directly in the onClick callback?

File: packages/shared/src/contexts/components/InjectedDialog.tsx [hunk 6]
```diff
@@ -74,6 +83,7 @@ export function InjectedDialog(props: InjectedDialogProps) {
         dialogCloseButton,
         dialogContent,
         dialogTitle,
+        dialogTitleTail,
```
Comment (Jack-Works) [inline packages/shared/src/contexts/components/InjectedDialog.tsx]: maybe rename to dialogTitleEndingContent

File: packages/dashboard/src/pages/Welcome/index.tsx [hunk 7]
```diff
@@ -33,7 +33,7 @@ const IFrame = styled('iframe')(({ theme }) => ({
 export default function Welcome() {
     const iframeRef = useRef<HTMLIFrameElement | null>(null)
     const mode = useTheme().palette.mode
-    const navigate = useNavigate()
+    const [, setupSocialNetwork] = useSetupSocialNetwork()
```
Comment (Jack-Works) [inline packages/dashboard/src/pages/Welcome/index.tsx]: never used.

File: packages/mask/src/social-network/ui.ts [hunk 8]
```diff
@@ -86,8 +86,10 @@ export async function activateSocialNetworkUIInner(ui_deferred: SocialNetworkUI.
     ui.collecting.currentVisitingIdentityProvider?.start(signal)
 
     ui.injection.pageInspector?.(signal)
-    if (Flags.toolbox_enabled) ui.injection.toolbox?.(signal)
-    ui.injection.setupPrompt?.(signal)
+    if (Flags.toolbox_enabled) {
```
Comment (Jack-Works) [inline packages/mask/src/social-network/ui.ts]: Looks like this PR contains many code from #6114. I'll defer the review.

File: packages/mask/src/extension/background-script/SocialNetworkService.ts [hunk 9]
```diff
@@ -22,6 +27,28 @@ function requestSNSAdaptorPermission(ui: SocialNetworkUI.Definition) {
     return requestExtensionPermission({ origins: [...ui.declarativePermissions.origins] })
 }
 
+function requestSNSAdaptorsPermission(uis: SocialNetworkUI.Definition[]) {
```
Comment (Jack-Works) [inline packages/mask/src/extension/background-script/SocialNetworkService.ts:30]: I suggest you to also remove `requestSNSAdaptorPermission`.

File: packages/mask/src/extension/background-script/SocialNetworkService.ts [hunk 10]
```diff
@@ -22,6 +27,28 @@ function requestSNSAdaptorPermission(ui: SocialNetworkUI.Definition) {
...
+    const req = uis.filter((x) => !x.permission?.request())
```
Comment (Jack-Works) [inline packages/mask/src/extension/background-script/SocialNetworkService.ts]: I made a check and found no SNS Adaptor is using `definition.permission.request`. I suggest to remove this field, and always use `declarativePermissions.origins`. That will allow us to skip the loading to the SNS Adaptor.

**Review by Jack-Works — CHANGES_REQUESTED** [review]
change request

File: packages/dashboard/src/pages/Welcome/index.tsx [hunk 11]
```diff
@@ -79,7 +79,10 @@ export default function Welcome() {
             iframeRef={iframeRef}
             privacyPolicyURL={agreementContentPageURL}
             iframeLoadHandler={handleIFrameLoad}
-            agreeHandler={() => Services.SocialNetwork.setupSocialNetwork('twitter.com')}
+            agreeHandler={async () => {
+                const url = await Services.SocialNetwork.setupSocialNetwork('twitter.com', false)
+                url && location.assign(url)
```
Comment (septs) [inline packages/dashboard/src/pages/Welcome/index.tsx]:
```suggestion
                if (url) location.assign(url)
```

File: packages/mask/src/components/GuideStep/index.tsx (hunk 2 above)
Comment (guanbinrui) [reply packages/mask/src/components/GuideStep/index.tsx]: ping

File: packages/mask/src/components/InjectedComponents/ToolboxUnstyled.tsx (hunk 4 above)
Comment (guanbinrui) [reply packages/mask/src/components/InjectedComponents/ToolboxUnstyled.tsx]: Recommend to use mask icon component. You can find here: `packages/mask/src/resources/MaskIcon.tsx`

File: packages/mask/src/social-network/ui.ts (hunk 3 above)
Comment (Lanttcat) [reply packages/mask/src/social-network/ui.ts:231]: We need to have the user authorize all SNS before creating persona.

File: packages/mask/src/extension/background-script/SocialNetworkService.ts [hunk 12]
```diff
@@ -28,10 +28,8 @@ function requestSNSAdaptorPermission(ui: SocialNetworkUI.Definition) {
 }
 
 function requestSNSAdaptorsPermission(uis: SocialNetworkUI.Definition[]) {
-    const req = uis.filter((x) => !x.permission?.request())
-    if (!req.length) return req
     return requestExtensionPermission({
-        origins: [...req.map((x) => x.declarativePermissions.origins).flat()],
+        origins: [...uis.map((x) => x.declarativePermissions.origins).flat()],
```
Comment (septs) [inline packages/mask/src/extension/background-script/SocialNetworkService.ts]:
```suggestion
        origins: uis.map((x) => x.declarativePermissions.origins).flat(),
```

File: packages/mask/src/components/GuideStep/index.tsx (hunk 2 above)
Comment (yanzhihong23) [reply packages/mask/src/components/GuideStep/index.tsx]: change url to location ? any difference ?

File: packages/mask/src/components/GuideStep/index.tsx [hunk 13]
```diff
@@ -119,6 +119,12 @@ export default function GuideStep({
...
+        url.pathname = '/home'
+        location.href = url.href
```
Comment (guanbinrui) [inline packages/mask/src/components/GuideStep/index.tsx]:
```suggestion
        location.assign(url.href)
```

**Review by guanbinrui — APPROVED** [review]

**Review by yanzhihong23 — APPROVED** [review]

File: packages/dashboard/src/pages/Welcome/index.tsx [hunk 14]
```diff
@@ -79,7 +77,10 @@ export default function Welcome() {
             iframeRef={iframeRef}
             privacyPolicyURL={agreementContentPageURL}
             iframeLoadHandler={handleIFrameLoad}
-            agreeHandler={() => navigate(DashboardRoutes.Setup)}
+            agreeHandler={async () => {
+                const url = await Services.SocialNetwork.setupSocialNetwork('twitter.com', false)
```
Comment (Jack-Works) [inline packages/dashboard/src/pages/Welcome/index.tsx:81]: Why do you set up Twitter by default?

**Pull request merged**